app/models/package.rb

```diff
@@ -0,0 +1,6 @@
+class Package < ApplicationRecord
+  enum delivery_status: { new: 'new', processing: 'processing', delivered: 'delivered', cancelled: 'cancelled' }, _prefix: :status
+
+  validates_uniqueness_of :tracking_number
+  validates_length_of :tracking_number, is: 12
+end
```

Comment:
This is old syntax of validation, not recommended by rubocop and style quide now
You can do something like
validates :tracking_number, length: { maximum: 100 }, uniqueness: true (example)